### ISSUE TITLE
1.9.0.0 - Alpha thread

### ISSUE THREAD

**Issue by DillonN** [nicehash/NiceHashMiner#1114]
Please post issues related to the 1.9.0.0 alpha here!

Known issues:
* Parallel benchmarking 
  * May not work properly with specific miners such as dtsm, Prospector and ClaymoreCryptonote_old. If having issues you can force benchmarks to run sequentially or manually benchmark. 
    * REMEMBER: For miner programs that haven't been updated, you can copy over your old benchmark files and they will already have speeds entered in your new install
  * On certain miners may immediately terminate if multiple start at the exact same time. Just retry and they should go through
  * The three dots that increment in the benchmark list on the current algo/miner may not update correctly

**Comment by Smeagol001** [comment]
![image](https://user-images.githubusercontent.com/13831373/38148760-118fa28c-3458-11e8-80a2-1d5ac1c98489.png)

somehow  tzhe app isnt  displaying the statistics anymore but looking on the cmd windows i can tell its working

**Comment by Pjalchemist** [comment]
Windows 10 x64 Pro, build 1709
Updates disabled 
Defender disabled

When downloading via Chrome on the laptop, Windows goes nuts about virus. 

Installing the miner no issues, when benching it crashed the moment it starts to bench. Have tried reboots, right click run as admin.

**Comment by JCarangoH** [comment]
There is an error in exchange value in 1.9.0.0 version, look:

1.9.0.0:
![captura de pantalla 2018-03-30 a la s 3 53 12 p m](https://user-images.githubusercontent.com/11586954/38153513-cd4ce5a0-3432-11e8-8dab-64b1463ff362.png)

1.8.1.10:
![captura de pantalla 2018-03-30 a la s 3 54 07 p m](https://user-images.githubusercontent.com/11586954/38153526-ddfb7254-3432-11e8-9eaa-606f594b42c3.png)

Both screenshots were taken at same moment...

**Comment by JCarangoH** [comment]
Forget about it, just have to reboot the pc, now it shows correctly.

**Comment by MaKCuMyC** [comment]
@DillonN why you ignore right name of DSTM equihash miner? it is D**ST**M, not D**TS**M.

**Comment by nerdatwork** [comment]
Why isnt the list of miner names screen **resizable** yet ? It has been forever and tedious to just scroll left and right when you got a huge screen.

**Comment by paulh69** [comment]
seems to have problems identifying or indexing devices?

**Comment by paulh69** [comment]
not sure how useful power setting would be unless set automatically, difficult to access settings from benchmarking screen and it seems doubtful that the power efficient algorithms would pay enough to make it worthwhile

**Comment by luky76** [comment]
hi, with new alpha version my system reset it when i try to benchmark somes of algorithms. only Equihash and Neoscrypt did successful. Whith old version work ok.
i got w 10 , ssd  ,  gtx 1070 and 650 w psu.

**Comment by paulh69** [comment]
something is causing a crash and it wipes the contents of all .json files?  they’re all still there but empty.

**Comment by luky76** [comment]
benchmark algorithms one by one, just one at a time.
no crash, no stop , no errors.
all ok and work

**Comment by Smeagol001** [comment]
i suggest making it possible to just specify how much load  on gpu/cpu can be allocated to nicehash in general like i want atleast 2 cores from my cpu to be free fro use of other programs since otherwise my os starts to lag and i cant even use  the mouse same for gpu make it so that at least like 2-5% wont be used like with a chekbox or something for globally affecting all miners atleast if they support something like this since 1.9 i cant even do a spreadsheet while  runnning nicehash :(

**Comment by shinnsd** [comment]
I've been getting rejected shares (a lot) from cryptonight. appearently it's a compatibility issue between xmr stak and the nicehash server (no issue if I use cast xmr). Looks like there is a new version of xmr stak that addressed this issue:

https://github.com/fireice-uk/xmr-stak/releases/tag/2.4.1

but he mentioned “workaround” so I assume it's not a complete fix.

Can you please look into it? Thanks

**Comment by boby361** [comment]
My 280x won't complete either equihash benchmark but my 7950 completes them fine. Something is up.

**Comment by OiHeldeke** [comment]
1. Compared to 1.8.2.0 pre2 both xmr cpu miners seem to be slower, especially xmrig. Older version was able to deliver ~59 H/s per thread, new does 48-49. This is on my slowest CPU. On faster CPU-s difference is smaller, but still some percent slower.
Edit: xmr-stak performance looks fine, but xmrig is as I wrote.

2. If I go to Settings->Devices/Algorithms->Check All Profitability (or check single), then it tries to open page https://api.nicehash.com/?utm_source=NHM&p=calc&name=CUSTOM&speed22=0.00&nhmver=1.8.1.10&cost=1&power=1 that does not exist.

**Comment by paulh69** [comment]
also gets stuck on “Unable to get Nicehash profitability data.  If you are connected to the internet, try again later.”  does not retry automatically

**Comment by paulh69** [comment]
also gets stuck on “Unable to get Nicehash profitability data.  If you are connected to the internet

**Comment by paulh69** [comment]
benchmarks don’t always run, comes up with “benchmarks successful ok” several times

**Comment by AtlAntA118** [comment]
Not really related to this specific release but it would be such an awesome feuture to right click on an algorythm and select "disable/enable on all cards"

**Comment by Smeagol001** [comment]
suggestion make all the collumns in the listviews sortable

**Comment by Smeagol001** [comment]
all sgminer benchmarks exept x11 fail and want me to use  precise mode despie  having used precise mode
im on a amd rx580 and ryzen5 1600x

**Comment by Spin55555** [comment]
Do you plan to implement the latest version of Claymore Dual Ethereum+ GPU Miner with keccak algorithm support? At the moment it is the most profitable combination.

**Comment by techman05** [comment]
Maybe its been mentioned but the stop button doesn't stop all. I went to benchmark after it running for a while and the xmr cpu app kept restarting even with the window closed and had to kill a nicehash process to stop it.

Also probably mentioned but the cryptoV7 is in cpu and gpu but only the cpu is running it. I'd hope if the miner is implemented I could somehow trigger it for the gpu.

Thanks

**Comment by pbuva** [comment]
1 BTC = N/A USD
often applications do not download data from the server and then evaluate mining as unprofitable.

EDIT: After 5 days, no answer, no sollution, still cant get profitability data for all rigs.

**Comment by jmhickman** [comment]
I've had great results with the software so far. 
Since we're in a testing phase, can we add the newest CastXMR to the suite? The latest version supports vastly higher cryptonight/v7 rates _without_ having to use the stupid blockchain driver.
http://www.gandalph3000.com/cast_xmr/cast-xmr-optimized-cryptonight-miner-for-rx-vega/

**Comment by bmassy** [comment]
Cannot Mine Cryptonightv7 with AMD cards. "AMD device not found, Backend AMD disabled, No miner backend enabled"
But mines perfectly fine on my GTX 1050ti..

**Comment by bmassy** [comment]
So I deleted the bin folder for that miner so nicehash would re-download and setup the miner again and it worked.

**Comment by CryptoVaraan** [comment]
@DillonN I see many comments here, but no reply from you? While i wonder if it helps to post more...
The displayed value in EUR is 10 times too high for **Global rate** and **Balance** in this version...
Have a nice day :)

**Comment by YuriyNa** [comment]
Разработчик похоже забросил проект. Жаль, на Найсхеше теперь на AMD не помайнишь((((.

**Comment by YuriyNa** [comment]
Алгоритм NeoScrypt на картах AMD не работает. Не проходит бенчмарк, при запуске вручную окно майнера открывается и сразу закрывается. Проверено на разных картах на трех системах.

**Comment by Bathmat** [comment]
I am getting a lot (5-10%) of this error: `AMD Invalid Result GPU ID 1` when using xmr-stak for V7.

This issue was brought up here: https://github.com/fireice-uk/xmr-stak/issues/1337

And is potentially resolved by: https://github.com/fireice-uk/xmr-stak/pull/1437

So perhaps releasing a newer version of xmr-stak might solve the issue.

**Comment by dandawn13** [comment]
Whenever I put value to electricity cost, Nicehash will only mine 1 GPU and shows the net profit of that. Other GPUs will not work until I put the electricity cost back to 0.

**Comment by DillonN** [comment]
Sorry for the delays everyone, April was busy. 

@Smeagol001 has this been an intermittent problem? If I could get the log file for a period when this happened could help greatly. 

@JCarangoH this issue has been fixed in code and will be in next update. Was related to differing decimal separators (comma/period)

@MaKCuMyC actually surprised I had missed that, brain fart there

@nerdatwork size of the main window is managed by code and difficult to make resizeable without a lot of effort. I probably won't be doing this soon because in the future I want to overhaul the main window somewhat (that is a bit further out)

@Smeagol001 intensity is determined by the miner programs and each one has their own way of doing it. A way to edit this from within NHML in a way that it abstracts all these different settings has been talked about but is not currently planned. Also the wiki currently leaves much to desire in explaining this. As it stands you can specify how many threads xmr-stak is using by editing the file `bin\xmr-stak\cpu.txt` and removing thread entries until you have the amount you want to use. Each thread entry looks like `    { "low_power_mode" : false, "no_prefetch" : true, "affine_to_cpu" : 10 },` but will have different values for `affine_to_cpu` which determines which physical thread they bind to (if you have HyperThreading make sure to only use every other value so you get a physical core and not an HT).

@shinnsd this is indeed due to an incompatibility with xmr-stak and NH servers, there is a more robust fix in their newest version which will be included in next release

@OiHeldeke 1. CNV7 still needs some time to catch up but they are making it better with every release. 2. The pages were changed server side and new links will be in the next version

@AtlAntA118 that is an interesting idea. I've been thinking of implementing groups for devices so that you only manage one list of algorithms/speeds (e.g. if you have identical cards) so if that is your use case that would help. 

@Smeagol001 I am looking into sgminer instability, which seams to be very bad when remoting into the computer. This is a known issue in sgminer.

@Spin55555 yes

@jmhickman we are in the fix-and-don't-add part of this test phase but can certainly look into CastXMR next

@dandawn13 mining on specific GPUs will not commence if their profit is 0 or negative after deducting the estimated power costs

In general:

Problems with benchmarking/miners not stopping are attributed to one of the known issues which is fixed

Problems with connectivity are not exclusive to this release and due to a huge number of factors. We continue to strive to make these issues as little as possible on the NHML side

**TLDR**

* Benchmarking is mostly fixed

* CNV7 will be fixed with new miners

* Exchange rate problems are fixed

* ClaymoreDual w/ keccak will be supported

* sgminer is still problematic (not new to this rel)

**Comment by ryangray0** [comment]
RE: Power consumption / profit: This is going to be a great feature.

Here is what I see: System benchmarks and runs fine with no values in for energy cost, or energy consumption in algos. When I add an energy cost in settings and populate all algo enrgy consumption values for all cards, no miner initiates when I click 'start'. If I remove the energy consumption value from any algo, that algo will initiate when I click 'start'. If I leave all energy consumption values in algos, and change energy cost to 0, system runs normally.

I apologize if this was covered in another post, this is my first contribution to the effort. Great product!

**Comment by DillonN** [comment]
@ryangray0 there is currently no auto-filling in for power usage - you must fill it all in manually.

When you hit start mining, NHML searches for the most profitable algorithm after subtracting power costs based on the values you entered for power usage and price per kilowatt-hour. If no enabled algos are above 0 profit after this subtraction then none will start

If you think the profits should be above 0 then you can forward me the logs and I can take a look to see if something is up

**Comment by dandawn13** [comment]
@DillonN we're talking about the new feature "electricity cost" and not the profit margin. It doesn't work on multiple gpu. Only works when a single gpu is enabled.

**Comment by clizmax** [comment]
@DillonN why in the version of nhml 1.9.0.0 alpha at the start of the program pops up the window "can not get nicehash profit data". this began to appear when version 1.9.0.1 was released. especially if you did so ??
and I do not want to switch to the new version, because Miner xmrig was removed, it yields 6 times more than xmr-stak ... return everything as it was, please

**Comment by S74nk0** [comment]
Issue related to old nhm version/build. Please use latest version and open a new issue for any bugs or feature requests.
